File: app-exemplo1/src/main/java/com/github/fabriciolfj/appexemplo1/repository/UserRepository.java
package com.github.fabriciolfj.appexemplo1.repository;

import com.github.fabriciolfj.appexemplo1.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, Integer> {

    Optional<User> findUserByUsername(String u);
}

File: hands-on/src/main/kotlin/com/github/fabriciolfj/handson/domain/repository/UserRepository.kt
package com.github.fabriciolfj.handson.domain.repository

import com.github.fabriciolfj.handson.domain.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface UserRepository : JpaRepository<User, String> {

    fun findByUsername(username: String) : Optional<User>
}